gempython/tools/optohybrid_user_functions_uhal.py

```diff
@@ -696,12 +700,20 @@ def getScanResults(device, gtx, numpoints, debug=False):
 
     return results
 
-def getUltraScanResults(device, gtx, numpoints, debug=False):
+def getUltraScanResults(device, gtx, numpoints, debug=False, numtrigs=1000, isLatency=False):
     scanBase = "GEM_AMC.OH.OH%d.ScanController.ULTRA"%(gtx)
+    ohnL1A_0 = getLinkL1Acount(device,gtx)
+    ohnL1A = getLinkL1Acount(device,gtx)
     while (readRegister(device,"%s.MONITOR.STATUS"%(scanBase)) > 0):
         msg = "%s: Ultra scan still running (0x%x), not returning results"%(device,
                                                                             readRegister(device,"%s.MONITOR.STATUS"%(scanBase)))
         ohlogger.debug(colormsg(msg,logging.DEBUG))
+        if (isLatency):
+            if ((getLinkL1Acount(device,gtx)-ohnL1A) > numtrigs):
+                print "At link %s: %d/%d L1As processed, %d%% done" %(gtx, getLinkL1Acount(device,gtx)-ohnL1A_0, numpoints*numtrigs, (getLinkL1Acount(device,gtx)-ohnL1A_0)*100./(numpoints*numtrigs))
```

Comment:
* You can get `numtrigs` from the scan config
* I'm be way more in favour of simply printing this out if `debug` is `True` rather than adding more arguments to the function
* Did you test that this behaves in a reasonable way?  The high rate behaviour is what I'm dubious of.